drivers/wasapi/audio_driver_wasapi.cpp

```diff
@@ -463,36 +421,160 @@ Error AudioDriverWASAPI::audio_device_init(AudioDeviceWASAPI *p_device, bool p_i
 		hr = device_audio_client_3->InitializeSharedAudioStream(0, period_frames, pwfex, nullptr);
 		if (hr != S_OK) {
 			print_verbose("WASAPI: InitializeSharedAudioStream failed with error 0x" + String::num_uint64(hr, 16) + ", falling back to IAudioClient.");
-			CoTaskMemFree(pwfex);
-			return audio_device_init(p_device, p_input, p_reinit, true);
+			return S_FALSE; // Signal to fall back to IAudioClient
+		}
+	}
+
+	return hr;
+}
+
+// Helper function to setup buffer and latency
+void AudioDriverWASAPI::setup_buffer_and_latency(AudioDeviceWASAPI *p_device, bool p_input, bool use_client3, WAVEFORMATEX *pwfex) {
```

Comment:
Should not return void, since we have no way to propagate errors out of this function.